packages/std-client/src/setup/setupMUDV2Network.ts

```diff
@@ -9,36 +9,35 @@ import {
   SingletonID,
 } from "@latticexyz/network";
 import { BehaviorSubject, concatMap, from, Subject } from "rxjs";
-import { Components, createWorld, defineComponent, Type } from "@latticexyz/recs";
+import { Components, defineComponent, Type, World } from "@latticexyz/recs";
 import { computed } from "mobx";
 import { keccak256, TableId } from "@latticexyz/utils";
 import { World as WorldContract } from "@latticexyz/world/types/ethers-contracts/World";
 import { abi as WorldAbi } from "@latticexyz/world/abi/World.json";
 import { defineStringComponent } from "../components";
-import { ContractComponent, SetupContractConfig } from "./types";
+import { ContractComponent, ContractComponents, SetupContractConfig } from "./types";
 import { applyNetworkUpdates, createEncoders } from "./utils";
-import { defineStoreComponents, EntityID } from "@latticexyz/recs";
-import storeMudConfig from "@latticexyz/store/mud.config.mjs";
-import worldMudConfig from "@latticexyz/world/mud.config.mjs";
-import { MUDUserConfig } from "@latticexyz/cli";
+import { EntityID } from "@latticexyz/recs";
+import { defineContractComponents as defineStoreComponents } from "../mud-definitions/store/contractComponents";
+import { defineContractComponents as defineWorldComponents } from "../mud-definitions/world/contractComponents";
 
-type SetupMUDV2NetworkOptions<M extends MUDUserConfig = MUDUserConfig> = {
-  mudConfig: M;
+type SetupMUDV2NetworkOptions<C extends ContractComponents> = {
   networkConfig: SetupContractConfig;
+  world: World;
+  contractComponents: C;
   initialGasPrice?: number;
   fetchSystemCalls?: boolean;
   syncThread?: "main" | "worker";
 };
 
-export async function setupMUDV2Network<M extends MUDUserConfig = MUDUserConfig>({
-  mudConfig,
+export async function setupMUDV2Network<C extends ContractComponents>({
   networkConfig,
+  world,
+  contractComponents,
   initialGasPrice,
   fetchSystemCalls,
   syncThread,
-}: SetupMUDV2NetworkOptions<M>) {
-  const world = createWorld();
```

Comment:
ooc why is it important to pass this in as an argument instead of creating here?

Reply:
I need world to create contractComponents to pass them to setup, so world has to come b4 setup